app/views/course/courses/show.html.slim

```diff
@@ -15,13 +15,15 @@
                 collection: current_course.managers.includes(:user).map(&:user)
 
   - if current_course.user?(current_user) || can?(:manage, current_course)
-    h2 = t('.announcements')
-    div.message-holder
-      = render partial: current_course.announcements.currently_active, spacer_template: 'announcements/announcement_spacer'
+    - if !current_course.announcements.currently_active.empty?
```

Comment:
Cache the query result to a variable ? 

Otherwise there are 2 queries generated if there're any announcements.